quandl/model/merged_dataset.py

```diff
@@ -0,0 +1,150 @@
+        if not isinstance(dataset_codes, list):
+            raise ValueError('dataset codes must be specified in a list')
+        # column_index is handled by individual dataset get's
+        if 'params' in options:
+            options['params'].pop("column_index", None)
+        datasets = list([cls._get_dataset(dataset_code, **options)
+                         for dataset_code in dataset_codes])
+        return MergedDataset(None, datasets)
+
+    @classmethod
+    def _get_dataset(cls, dataset_code, **options):
+        options_copy = options.copy()
+        # data_codes are tuples e.g., ('GOOG/NASDAQ_AAPL', {'column_index":
+        # [1,2]}) or strings 'NSE/OIL'
+        code = cls._get_request_dataset_code(dataset_code)
+
+        dataset = Dataset.get(code, **options_copy)
+        dataset.column_index = []
+        # ensure if column_index dict is specified, value is a list
+        params = cls._get_request_params(dataset_code)
+        if 'column_index' in params:
+            column_index = params['column_index']
+            if not isinstance(column_index, list):
+                raise ValueError(
+                    "%s : column_index needs to be a list of integer indexes" % code)
+            dataset.column_index = column_index
+        # if column_indexes are specified, check that they actually exist
+        max_column_index = len(dataset.column_names) - 1
+        for index in dataset.column_index:
+            if index > max_column_index or index < 1:
+                raise ValueError("%s : requested index %s is out of range. \
+                                  Min index is 1 and Max index is %s" % (code, index,
+                                                                         max_column_index))
+        return dataset
+
+    @classmethod
+    def _get_request_dataset_code(cls, dataset_code):
+        if isinstance(dataset_code, tuple):
+            return dataset_code[0]
+        elif isinstance(dataset_code, string_types):
+            return dataset_code
+        else:
+            raise ValueError("Arguments in list must be tuple or string")
+
+    @classmethod
+    def _get_request_params(cls, dataset_code):
+        if isinstance(dataset_code, tuple):
+            return dataset_code[1]
+        return {}
+
+    @property
+    def column_names(self):
+        return list(unique_everseen(
+            [self._rename_columns(dataset, column_name) if index > 0 else column_name
+             for dataset in self._datasets
+             for index, column_name in
+             enumerate(dataset.raw_data['column_names']) if self._include_column(dataset, index)]))
+
+    @property
+    def oldest_available_date(self):
+        return min(self._get_dataset_attribute('oldest_available_date'))
+
+    @property
+    def newest_available_date(self):
+        return max(self._get_dataset_attribute('newest_available_date'))
+
+    def data(self, **options):
+        dataset_data_list = [dataset.data(**options)
```

Comment:
actually it looks like options is always copied cause it is passed in as **options, no need to make a copy of each options, it is already done internally